src/components/Content.tsx

```diff
@@ -43,15 +42,15 @@ const mapStateToProps = (state: State) => {
   const tutorialStep = isLoading ? tutorialStepLocal : +(getSetting(state, 'Tutorial Step') ?? 1)
 
   // do no sort here as the new object reference would cause a re-render even when the children have not changed
-  const rootThoughts = getThoughts(state, [ROOT_TOKEN]).filter(({ value, rank }) => showHiddenThoughts || isChildVisible(state, [value], { value, rank }))
+  const rootThoughtsLength = (showHiddenThoughts ? getThoughts(state, [ROOT_TOKEN]) : getThoughts(state, [ROOT_TOKEN]).filter(({ value, rank }) => isChildVisible(state, [value], { value, rank }))).length
 
   return {
     focus,
     search,
     showModal,
     isTutorial,
     tutorialStep,
-    rootThoughts,
+    rootThoughtsLength,
```

Comment:
Won't this prevent a re-render when a thought has been edited?

Reply:
It will re-render if the edit causes change of total number of visible children of root context. I don't think we need to re-render `Content` if edit doesn't cause visible children to change.

Reply:
If other children change, such as from another data source, then we will need to re-render.

Reply:
@shresthabijay

Reply:
> If other children change, such as from another data source, then we will need to re-render.

I don't understand. Can you explain briefly ?

My understanding is any changes in the store that cause number of visible child to change should re-render `Content`. Changes in other context should simply be handled by `Subthoughts` component.

Reply:
Oh, I guess you're right. I forgot that Subthoughts would take care of it. Or maybe I was thinking of before Subthoughts' `mapStateToProps` was changed.